src/services/chain.ts

```diff
@@ -352,23 +355,15 @@ export class ChainContext {
           this
         );
 
-        try {
-          await processBlock(this, block, events);
-
-          // Block height metric
-          this.registry.lastProcessedBlock = blockToRegistryBlock(block);
-          this.registry.write();
-          metrics.blockHeight.labels(chainId.toString()).set(blockNumber);
-        } catch {
-          log.error(`Error processing block ${blockNumber}`);
-        }
-
+        await processBlock(this, block, events);
         log.debug(`Block ${blockNumber} has been processed`);
       } catch (error) {
-        log.error(
-          `Error in pollContractForEvents for block ${blockNumber}`,
-          error
-        );
+        log.error(`Error processing block ${blockNumber}`, error);
+      } finally {
+        // Block height metric
+        this.registry.lastProcessedBlock = blockToRegistryBlock(block);
+        await this.registry.write();
+        metrics.blockHeight.labels(chainId.toString()).set(blockNumber);
```

Comment:
but if the block was not process, why should we write it is. 
Wouldn't this cover up next issue we have?